packages/fluxible-router/lib/handleHistory.js

```diff
@@ -121,14 +121,21 @@ function createComponent(Component, opts) {
         _onHistoryChange: function (e) {
             var props = this.props;
             var url = this._history.getUrl();
-            var currentRouteMap = props.currentRoute || {};
-            var currentUrl = currentRouteMap.url;
+            var currentRoute = props.currentRoute || {};
+            var nav = props.currentNavigate || {};
+
+            // Add currentNavigate.clientRelativeUrl for https://github.com/yahoo/fluxible/issues/349:
+            // "Safari popstate issue causing handleHistory.js to execute the navigateAction on page load".
+            // This needs app to dispatch "clientRelativeUrl" as part of the payload for the NAVIGATE_START event
+            // on server side.  For client side navigation, this field is not needed and will not be set.
+            var currentUrl = nav.clientRelativeUrl || currentRoute.url;
```

Comment:
@skippykawakami For example, if request url in browser is http://mydomain.com/abc, and ATS changes it to http://mydomain.com/abc?a=1 for whatever reason, (**NOTE** using query param is not recommended.  should use HTTP header instead.), your app can dispatch `NAVIGATE_START` event with payload like this:
```
{
   url: '/abc',
  method: 'GET',
  clientRelativeUrl: '/abc?a=1'
}
```